### PR TITLE
Feature/update british citizenship d365

### PR DESCRIPTION
**Pull request by grange85**
I think this also includes the changes from the general enquiry branch so - hopefully two birds...

Issues - 

* I'm not sure I have the british-citizenship javascript validation right - the form appears to work but not 100% certain that it is all as it should be
* The form seems to be submitting an unwanted newsletter request - I don't think this is related to the problems since it happens on dev-www with a previous version ... but if someone could figure that out?

This version is currently deployed on dev (slack me for the url)

The two forms that have been changed are
* General enquiries
* Request a search for British Citizenship

### CODE REVIEW

File: forms/form-british-citizenship.php [hunk 1]
```diff
@@ -283,8 +339,10 @@ function process_form_british_citizenship() {
 		// Error messages for individual form fields stored into an array
 		// IMPORTANT: $tna_error_messages array keys must match exactly the $form_fields array keys
 		$tna_error_messages  = array(
-			'Certificate holder name'   => 'Please enter the certificate holder’s name',
-			'Full name'                 => 'Please enter your full name',
+			'Certificate holder\'s first name'   => 'Please enter the certificate holder’s first name',
+			'Certificate holder\'s last name'   => 'Please enter the certificate first holder’s last name',
```
Comment (AshTNA) [inline forms/form-british-citizenship.php:343]: Is this a typo?

**Review by AshTNA — COMMENTED** [review]
Hi Andy,

Would you be able to provide some context on what the changes are?

The validation doesn't seem to work on the British citizenship form. I was able to submit a completely empty form.

The newsletter issue is probably something to do with the `subscribe_to_newsletter_post` function, it must be being passed a "Yes" value even when the user hasn't ticked that they want to subscribe. 

Thanks

**Comment by gtvj** [comment]
Hi @grange85 - I've just tried the British Citizenship form locally and it seems like the validation isn't working. I was able to proceed to the next step without having entered anything into any of the required fields. 

It looks like this might be because the `required` and `aria-required="true"` attributes aren't present. I've had a go at re-introducing them and this seems to have solved the problem.

**Comment by gtvj** [comment]
Hi @grange85 - I've just tried the British Citizenship form locally and it seems like the validation isn't working. I was able to proceed to the next step without having entered anything into any of the required fields. 

It looks like this might be because the `required` and `aria-required="true"` attributes aren't present. I've had a go at re-introducing them and this seems to have solved the problem.

File: forms/form-british-citizenship.php [hunk 2]
```diff
@@ -26,14 +26,23 @@ function return_form_british_citizenship( $content ) {
 	                        <p>Enter the details of the certificate holder at the time the certificate was issued.</p>
 	                    </div>
 	                    <div class="form-row">
-	                        <label for="certificate_name">Full name</label>
-	                        <input type="text" id="certificate_name" name="certificate-name" aria-required="true" required ' . set_value( 'certificate-name' ) . '>
-	                        ' . field_error_message( 'certificate-name', 'Certificate holder name' ) . '
+	                        <label for="certificate_first_name">Certificate holder first name</label>
+	                        <input type="text" id="certificate_first_name" name="certificate-first-name" ' . set_value( 'certificate-first-name' ) . '>
```
Comment (gtvj) [inline forms/form-british-citizenship.php:30]: Should this field have the `required` and `aria-required="true"` attributes that were on the `certificate_name` field?

Comment (gtvj) [inline forms/form-british-citizenship.php:30]: Should this field have the `required` and `aria-required="true"` attributes that were on the `certificate_name` field?

**Review by gtvj — APPROVED** [review]

**Pull request merged**